src/types.md

```diff
@@ -0,0 +1,183 @@
+# Questions about your types
+
+## My 'class' needs references to lots of other things in order to do its job, and that conflicts with other things which also hold similar references. What do I do?
+
+It's common in C++ to have objects which need to mutate other objects to get
+their work done. Those other objects may be accessed by several different
+objects.
+
+```mermaid
+flowchart LR
+    subgraph Shared functionality
+    important[Important Shared Object]
+    end
+    subgraph ObjectA
+    methodA[Method]
+    refa[Mutable Reference]-->important
+    methodA-. Acts on shared object.->important
+    end
+    subgraph ObjectB
+    refb[Mutable Reference]-->important
+    methodB[Method]
+    methodB-. Acts on shared object.->important
+    end
+    main --> ObjectA
+    main --> ObjectB
+    main-. Calls .-> methodA
+    main-. Calls .-> methodB
+```
+
+You can't have multiple mutable references to a shared object, so what do you do?
```

Comment:
Maybe to make the contrast clear, add "In Rust", i.e.: "In Rust, you can't have multiple mutable references..."